### PR TITLE
Cleanup Google Sign-In settings

### PR DESCRIPTION
**Pull request by ShivanKaul**
- Don't use kGoogleLoginControlType to control Google Sign-In feature; now it's purely a content setting.
- Clarify text for default/allow-exception/block-exception for Google Sign-In content setting.

Resolves https://github.com/brave/brave-browser/issues/28358

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

**OLD PERMISSION TEXT:**

image

**NEW: PERMISSION DOES NOT EXIST:**

image

**OLD SITE PERMISSION TEXT:**
image

**NEW SITE PERMISSION TEXT**

image

### CODE REVIEW

File: app/brave_settings_strings.grdp [hunk 1]
```diff
@@ -548,13 +548,16 @@
     Google Sign-In
   </message>
   <message name="IDS_SETTINGS_SITE_SETTINGS_GOOGLE_SIGN_IN_ASK" desc="Label for G sign-in site settings.">
-    Ask when a site wants to use legacy Google Sign-In
+    Sites can ask to use legacy Google Sign-In via third-party cookies
   </message>
   <message name="IDS_SETTINGS_SITE_SETTINGS_GOOGLE_SIGN_IN_BLOCK" desc="Label for G sign-in site settings.">
-    Block sites from using legacy Google Sign-In
+    Don't allow legacy Google Sign-In via third-party cookies
   </message>
```
Comment (petemill) [inline app/brave_settings_strings.grdp:555]: Can this message be removed now that it's not being used?

Comment (ShivanKaul) [reply app/brave_settings_strings.grdp:555]: We use it in `browser/resources/settings/brave_overrides/site_settings_page.ts` for the label to be displayed 
image

File: browser/resources/settings/brave_overrides/privacy_page.ts [hunk 2]
```diff
@@ -52,10 +52,10 @@ function InsertGoogleSignInSubpage (
       } else {
         googleSignInDefault.setAttribute(
           'toggle-off-label',
-          I18nBehavior.i18n('siteSettingsGoogleSignInBlock'))
+          I18nBehavior.i18n('siteSettingsGoogleSignInAsk'))
```
Comment (petemill) [inline browser/resources/settings/brave_overrides/privacy_page.ts:55]: This is just a side note (and TODO to fix later, cc @emerick as it looks like you may have edited this file most often recently) as it's the first time I've really looked in to this override file recently, but it's a little strange that we're creating some html elements via `getTrustedHtml` and then using `querySelector` to go through it and change attributes. The `settings-privacy-page` component _does_ have the `I18nMixin` so I assume we can just write the i18n string names inside the html with polymer bindings, e.g. `toggle-on-label="$i18n{siteSettingsGoogleSignInAsk}"`

File: app/brave_settings_strings.grdp (hunk 1 above)
Comment (petemill) [reply app/brave_settings_strings.grdp:555]: Seems like it's still being used in `site_settings_page` - that intentional?

**Review by petemill — APPROVED** [review]
I'm not 100% clued in as to the details of the feature, but perhaps that's helped as we talked about how this would be an understandable UX for the users. Thanks for discussing over DM, I think this is a better start, and we can massage the default toggle inside content settings or bringing it out to a more prominent area as needed later.

File: app/brave_settings_strings.grdp (hunk 1 above)
Comment (ShivanKaul) [reply app/brave_settings_strings.grdp:555]: Yes, I believe we need some text indicating what the current default value is (ask or block) in `site_settings_page`.

**Comment by ShivanKaul** [comment]
The CI for Windows and Linux is failing because of an S3 upload issue: 
- https://ci.brave.com/job/pr-brave-browser-enhancement-google-sign-in-dont-check-google-sign-in-pref-linux/12/execution/node/464/log/
- https://ci.brave.com/job/pr-brave-browser-enhancement-google-sign-in-dont-check-google-sign-in-pref-windows-x86/5/execution/node/437/
- https://ci.brave.com/job/pr-brave-browser-enhancement-google-sign-in-dont-check-google-sign-in-pref-windows/5/execution/node/463/

**Pull request merged**